### DISCUSSION TITLE
AppImage support for Linux OSes

### DISCUSSION THREAD

**Discussion by ihhub** [ihhub/fheroes2#7738] [Ideas]
As proposed in Discord AppImage support might be useful for Linux OSes. This is a topic for discussion about a potential of this package support, possible range of problems and how to implement this.

**Comment by oleg-derevenetz** [comment]
I do not yet understand how, in the case of packaging in the AppImage format, separate scripts/utilities used to download the demo version and unpack the assets of the original game should work. Apparently, it will still need to be packed into a ZIP archive, and the advantage of AppImage will only be that it will carry with it a certain set of libraries (SDL, SDL_mixer)?

In addition, SDL_mixer uses the plug-in mechanism to play various music formats (it can load the corresponding libs dynamically, like with FluidSynth - if there is FluidSynth library installed, then everything is fine, otherwise it will use the native MIDI backend), and I strongly doubt that the utility for packaging libraries into AppImage will cope with this, it will most likely need to be handled manually.

Also, what will the audience be like? I think most of them still use package systems that are "native" to the Linux distribution used (apt, yum, portage, etc).

**Comment by kenderipa** [comment]
I as a user would love it.
